packages/datetime/src/dateRangeInput.tsx

```diff
@@ -428,9 +662,12 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private getInputDisplayString = (boundary: DateRangeBoundary) => {
         const { values } = this.getStateKeysAndValuesForBoundary(boundary);
-        const { isInputFocused, inputString, selectedValue } = values;
+        const { isInputFocused, inputString, selectedValue, hoverString } = values;
 
-        if (isInputFocused) {
+        if (hoverString != null && !this.isControlled()) {
```

Comment:
Wasn't sure exactly how to handle hover values in controlled mode. You can still see the visual hovered range in the `DateRangePicker`; maybe it makes sense to show the hovered-range values in the fields as well.